### PR TITLE
[INTER-3588] Add style options for fastlane components

### PR DESCRIPTION
**Pull request by kfalkowski11**
#### Overview

Adding Fastlane style options when initializing component

### CODE REVIEW

File: src/types/fastlane.ts [hunk 1]
```diff
@@ -97,4 +97,55 @@ export interface IFastlaneInstance {
...
+    'DISCOVER',
+    'JCB',
+    'UNION-PAY',
+    'MAESTRO',
+    'ELO',
+    'MIR',
+    'HIPER',
+    'HIPERCARD',
+}
+
```
Comment (BoldCole) [inline src/types/fastlane.ts]: Don't use enum. It doesn't act as a purely virtual type and actually generates code that breaks tree shaking. Use a union type instead

```suggestion

export type IFastlaneCardBrandEnum = 
  'VISA' |
  'MASTERCARD' |
  'AMERICAN-EXPRESS' |
  'DINERS-CLUB' |
  'DISCOVER' |
  'JCB' |
  'UNION-PAY' |
  'MAESTRO' |
  'ELO' |
  'MIR' |
  'HIPER' |
  'HIPERCARD';
```

File: src/types/fastlane.ts [hunk 2]
```diff
@@ -97,4 +97,55 @@ export interface IFastlaneInstance {
...
+export interface IFastlaneAddressOptions {
+    // default: empty array = all locations allowed
+    allowedLocations: [string];
+}
+
+export interface IFastlaneCardOptions {
+    // default: empty array = all brands allowed
+    allowedBrands: [IFastlaneCardBrandEnum];
```
Comment (BoldCole) [inline src/types/fastlane.ts]: This type syntax describes a field that takes an array with only one element that is of "type" `IFastlaneCardBrandEnum`. Is this correct or is this field an array of "type" `IFastlaneCardBrandEnum`?

```suggestion
    allowedBrands: IFastlaneCardBrandEnum[];
```

File: src/types/fastlane.ts [hunk 3]
```diff
@@ -97,4 +97,55 @@ export interface IFastlaneInstance {
...
+
+export interface IFastlaneOptions {
+    shippingAddressOptions: IFastlaneAddressOptions,
+    cardOptions: IFastlaneCardOptions,
+    styles: IFastlaneStyleOptions
+}
+
+export interface IFastlaneAddressOptions {
+    // default: empty array = all locations allowed
+    allowedLocations: [string];
```
Comment (BoldCole) [inline src/types/fastlane.ts]: This type syntax describes a field that takes an array with only one element that is of "type" `string`. Is this correct or is this field an array of "type" `string`?

```suggestion
    allowedLocations: string[];
```

**Review by jeremywiebeofbold — APPROVED** [review]

File: src/types/fastlane.ts (hunk 1 above)
Comment (kfalkowski11) [reply src/types/fastlane.ts]: changed

File: src/types/fastlane.ts (hunk 2 above)
Comment (kfalkowski11) [reply src/types/fastlane.ts]: I was just copying Paypal's docs, but you're right, it should be `IFastlaneCardBrandEnum[]`. Updated.

File: src/types/fastlane.ts (hunk 3 above)
Comment (kfalkowski11) [reply src/types/fastlane.ts]: updated.

**Review by duke605 — APPROVED** [review]

**Review by BoldCole — APPROVED** [review]

**Pull request merged**